uniflow-core/src/main/kotlin/io/uniflow/core/flow/data/UIEvent.kt

```diff
@@ -23,7 +23,29 @@ package io.uniflow.core.flow.data
 open class UIEvent : UIData {
     object Loading : UIEvent()
     object Success : UIEvent()
-    data class Fail(val message: String? = null, val error: Throwable? = null, val state: UIState? = null) : UIEvent()
+    data class Error(val message: String? = null, val error: Throwable? = null, val state: UIState? = null) : UIEvent() {
+        override fun equals(other: Any?): Boolean {
+            if (this === other) return true
+            if (javaClass != other?.javaClass) return false
+
+            other as Error
+
+            if (message != other.message) return false
+            if (error?.javaClass != other.error?.javaClass) return false
+            if (error?.message != other.error?.message) return false
```

Comment:
don't have a good solution here 🤷‍♂️
Exception are not comparable. I know, I could add cause also 👍

Reply:
I'm just wondering why we are doing that?, could you provide more details

Reply:
Because the fhe following code isn't working: 

`
val error1 = IllegalStateException("boom")
val error2 = IllegalStateException("boom")
assert(error1 == error2)
`

then, when you assert about failed state or event, it will always fail because an Exception is not comparable :/

Reply:
You could add an additional check before these for `error === other.error`.
And for `error?.cause?.javaClass == other?.error?.cause?.javaClass`.

Reply:
but we need to compare content here, not sure for complete equality of pointers here? 🤔